mmv1/third_party/terraform/resources/resource_bigquery_table.go

```diff
@@ -147,13 +139,34 @@ func bigQueryTableSchemaDiffSuppress(_, old, new string, _ *schema.ResourceData)
 	}
 	var a, b interface{}
 	if err := json.Unmarshal([]byte(old), &a); err != nil {
-		log.Printf("[DEBUG] unable to unmarshal json - %v", err)
+		log.Printf("[DEBUG] unable to unmarshal old json - %v", err)
 	}
 	if err := json.Unmarshal([]byte(new), &b); err != nil {
-		log.Printf("[DEBUG] unable to unmarshal json - %v", err)
+		log.Printf("[DEBUG] unable to unmarshal new json - %v", err)
+	}
+
+	arrayA := a.([]interface{})
+	arrayB, ok := b.([]interface{})
+	if !ok {
+		return false
+	} else if len(arrayA) != len(arrayB) {
+		return false
+	}
+
+	// Sort the first level of the schema by name so reordering the
```

Comment:
A field can have fields nested inside of it [[schema](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#tablefieldschema)], so we still need to sort within the JSON function still, right? With this change reordering nested `fields` fields won't work as I'm reading it.

Reply:
Thanks, I didn't know about nested fields! I'll make this support both.